File: src/commonMain/kotlin/_08_piping/piping.kt
// Functions: piping, declared own sum
package _08_piping

data class Item(val title: String, val price: Int)
data class Order(val orderId: Int, val items: List<Item>)

fun totalRevenueOfImportantOrders(orders: List<Order>): Int =
    orders.filter { order -> order.items.any { it.price > 5 } }
        .flatMap { it.items }
        .map { it.price }
        .reduce { x, y -> x + y }

// Any questions?

File: src/commonMain/kotlin/_08_piping/piping.py
class Item:
    def __init__(self, title: str, price: int):
        self.title = title
        self.price = price


class Order:
    def __init__(self, order_id: int, items: [Item]):
        self.order_id = order_id
        self.items = items


def total_revenue_of_important_orders(orders: [Order]):
    return sum(
        [item.price
         for order in filter(lambda order: any([item.price > 5 for item in order.items]), orders)
         for item in order.items]
    )


def total_revenue_of_important_orders_mutable_variables(orders: [Order]):
    orders = filter(lambda order: any([item.price > 5 for item in order.items]), orders)
    prices = [item.price for order in orders for item in order.items]
    return sum(prices)


print(total_revenue_of_important_orders([Order(1, [Item("Banana", 6)])]))
